### ISSUE TITLE
Bug with Time.add_days

### ISSUE THREAD

**Issue by fclm1000** [vlang/v#3373]
**V version:** 
V 0.1.24 9861b24

**OS:** 
Linux Mint 18.3

**What did you do?**
```
module main

import (
    time
)

fn main() {
    a := time.parse_iso('Thu, 12 Dec 2019 06:07:45 GMT')
    b := a.add_days(1)  
    //
    println(a.format()) // prints 2019-12-12 06:07
    println(b.format()) // prints 2019-12-13 09:07 // more 1 day (ok) // more 3 hours (bug)
}
```
**What did you expect to see?**
2019-12-13 06:07
 
**What did you see instead?**
2019-12-13 09:07

**Comment by changrui** [comment]
oh, God,  my result is:
```
2019-02-28 06:07
2019-02-29 06:07
```

**Comment by Contextualist** [comment]
In short, 1) `time.parse_iso` does not interpret timezone information; 2) `time.calc_unix` assumes local timezone.

https://github.com/vlang/v/blob/0d93eeb3fecb16e1f3e0c6c8820140e67e947eb0/vlib/time/time.v#L223-L256

`time.parse_iso` extracts the date and time information (but not timezone) and use `time.new_time ` to initialize a time struct. After filling out the date time info, unix timestamp is calculated using `C.mktime`, which assumes the date time info is in local timezone. (**Problematic**)

@fclm1000, your machine is probably in UTC−03:00 timezone, so the unix timestamp you get is representing `2019-12-12 09:07 (UTC)` (`2019-12-12 06:07 (UTC-03:00)`). (While `a.format()` here uses the original date time information.)

Now, `time.Time.add_days` generate a new Time struct with the unix timestamp + 86400s (1 day). The date time you see is in UTC. (i.e. `2019-12-13 09:07`)


To fix the problem:
1. `Time` struct needs a timezone property
2. `time.parse_iso` parses the timezone given, (and maybe `time.parse` assume local timezone?)
3. Use an alternative of `C.mktime` that we can pass in the timezone information.

**Comment by fclm1000** [comment]
```
import (
    time
)

fn main() { 
    works_fine()
    should_i_access_v_unix()
}

fn works_fine() {
    t := time.now()
    t2 := t.add_days(1)
    //
    println(t.format())  // prints 2020-01-08 21:26 // time when I executed 
    println(t2.format()) // prints 2020-01-09 21:26 // ok, 1 day later 
}

fn should_i_access_v_unix() {
    /*
        // from module time

        pub struct Time {
        pub:
	        year   int
	        month  int
	        day    int
	        hour   int
	        minute int
	        second int
	        unix int
        }
    */
    t := time.now()
    println(t)          // shows v_unix instead of unix
    println(t.unix)     // prints (when I executed) 1578519820
    println(t.v_unix)   // prints (when I executed) 1578519820
                        // v_unix is not a member of time.Time
}
```
@Contextualist 

Indeed my machine timezone is UTC−03:00.

I made another test, starting with time.now and that was ok.

I am confused about Time.unix and Time.v_unix as in the code above.

Thanks for your attention ;)

**Comment by spytheman** [comment]
Use .unix in v. v_unix is an implementation detail concerning the name of the generated C structure field (the reason is that `unix` is considered by v's compiler to be a C reserved word).

**Comment by medvednikov** [comment]
Not anymore, `unix` should be fine now.

**Comment by watzon** [comment]
Running into this with the tests on Fedora

**Comment by alexesprit** [comment]
Was fixed in #3390.
